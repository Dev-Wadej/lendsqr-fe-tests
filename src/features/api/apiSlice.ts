import {createApi,fetchBaseQuery} from '@reduxjs/toolkit/query/react';

interface IUsersData{
    createdAt:string;
    orgName:string
   email:string;
   id:string;
   userName:string;
   phoneNumber:string;
  }
interface IEachUser{
    accountNumber: string;
    email:string;
    profile: {
        firstName: string;
        lastName: string;
        phoneNumber: string;
        avatar:string;
    gender: string;
    bvn: string;
    }
    guarantor: {
        firstName: string,
        lastName: string,
        phoneNumber: string;
        
      },
      education: {
        level: string;
        employmentStatus: string;
        sector: string;
        duration: string;
        officeEmail: string;
        monthlyIncome: string[];
        loanRepayment: string;

      },
      socials: {
        facebook: string;
        instagram: string;
        twitter: string;
      },

}

export const apiSlice = createApi({
    reducerPath: 'api',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1' }),
    tagTypes: ['Users'],
    endpoints: (builder) => ({
        getUsers: builder.query<IUsersData[],boolean>({
            query: () => '/users',
            providesTags: ['Users'],
        }),
        getEachUser: builder.query<IEachUser,string>({
            query: (id) => `/users/${id}`,
            providesTags: ['Users'],
        }),
    }),
    
})

export const {
    useGetUsersQuery,useGetEachUserQuery
} = apiSlice