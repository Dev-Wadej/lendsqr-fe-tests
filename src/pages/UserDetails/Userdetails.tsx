import React from 'react'
import {useNavigate,useParams} from 'react-router-dom'

import MenuIcon from '@mui/icons-material/Menu';
import './userdetails.scss'

//=========================Custom Hooks====================================
import {useSidebar } from '../../context/ContextProvider'
import { useGetEachUserQuery} from'../../features/api/apiSlice'
//============================================================================

//=========================SVG's===============   ============  ===========
import {ReactComponent as ArrowBack} from '../../data/svgs/backarrow.svg'
import {ReactComponent as StarFilled} from '../../data/svgs/starfilled.svg'
import {ReactComponent as StarUnfilled} from '../../data/svgs/starunfilled.svg'
//============================================================================

const Userdetails = () => {

  const{sidebar,setSidebar}=useSidebar()
  const navigate= useNavigate()
  const {id} = useParams()

  const {data}=useGetEachUserQuery(id!)
  const handleOpenSideBar=()=>{
    setSidebar?.(true)
  }
  const handleNavigationUser=()=>{
    navigate('/dashboard/user')
  }
  
  return (
    <div className='user-details-section-wrapper'>
      <div onClick={handleOpenSideBar} className='user-hamburger'>
      { !sidebar ? <MenuIcon/> : <></>}
    
      </div>
    <section className='user-details-section'>
      <div onClick={handleNavigationUser} className='user-details-arrow-back' style={{cursor:'pointer'}}>
        <span><ArrowBack/></span>
        <span>Back to Users</span>
      </div>
      <div className='user-details-header'>
        <h3>User Details</h3>
        <dl>
          <button>BLACKLIST USER</button>
          <button>ACTIVATE USER</button>
        </dl>
      </div>
      <div className='user-top-details-wrapper'>
      <section className='user-details-top'>
          <aside className='user-details-avatar-wrapper'>
            <span className='avatar-img'> 
            <img src={data?.profile.avatar} alt="User Avatar" />
            </span>
            <div>
              <h4>{data?.profile.firstName} {data?.profile.lastName}</h4>
              <span>{data?.accountNumber}</span>
            </div>
          </aside>
          <aside className='user-tier'>
            <h3>User's Tier</h3>
            <span>
              <StarFilled/>
              <StarUnfilled/>
              <StarUnfilled/>
            </span>
          </aside>
          <aside className='user-numerics'>
            <h4>₦200,000.00</h4>
            <p>9912345678/Providus Bank</p>
          </aside>
      </section>
      <section className='user-tab'>
        <button className='active'>General Details</button>
        <button>Documents</button>
        <button>Bank Details</button>
        <button>Loans</button>
        <button>Savings</button>
        <button>App and System</button>
      </section>
      </div>
      <section className='user-heavy-info'>
        <aside className='user-personal-info'>
          <h5>Personal Information</h5>
          <div>
          <dl>
            <div className='user-name'>full Name</div>
            <div>{data?.profile.firstName} {data?.profile.lastName}</div>
          </dl>
          <dl>
            <div className='user-name'>Phone Number</div>
            <div>{data?.profile.phoneNumber}</div>
          </dl>
          <dl>
            <div className='user-name'>Email Address</div>
            <div>{data?.email}</div>
          </dl>
          <dl>
            <div className='user-name'>Bvn</div>
            <div>{data?.profile.bvn}</div>
          </dl>
          <dl>
            <div className='user-name'>Gender</div>
            <div>{data?.profile.gender}</div>
          </dl>
          <dl>
            <div className='user-name'>Marital status</div>
            <div>Single</div>
          </dl>
          <dl>
            <div className='user-name'>Children</div>
            <div>None</div>
          </dl>
          <dl>
            <div className='user-name'>Type of residence</div>
            <div>Parent’s Apartment</div>
          </dl>
          </div>
        </aside>
        <hr/>
        <aside className='user-education-job'>
        <h5>Education and Employment</h5>
          <div>
          <dl>
            <div className='user-name'>level of education</div>
            <div>{data?.education.level}</div>
          </dl>
          <dl>
            <div className='user-name'>employment status</div>
            <div>{data?.education.employmentStatus}</div>
          </dl>
          <dl>
            <div className='user-name'>sector of employment</div>
            <div>{data?.education.sector}</div>
          </dl>
          <dl>
            <div className='user-name'>Duration of employment</div>
            <div>{data?.education.duration}</div>
          </dl>
          <dl>
            <div className='user-name'>office email</div>
            <div>{data?.education.officeEmail}</div>
          </dl>
          <dl>
            <div className='user-name'>Monthly income </div>
            <div>₦{data?.education.monthlyIncome[0]} - ₦{data?.education.monthlyIncome[1]} </div>
          </dl>
          <dl>
            <div className='user-name'>loan repayment</div>
            <div>{data?.education.loanRepayment}</div>
          </dl>
          </div>
        </aside>
        <hr/>
        <aside className='user-socials'>
        <h5>Socials</h5>
          <div>
          <dl>
            <div className='user-name'>Twitter</div>
            <div>{data?.socials.twitter}</div>
          </dl>
          <dl>
            <div className='user-name'>Facebook</div>
            <div>{data?.socials.facebook}</div>
          </dl>
          <dl>
            <div className='user-name'> Instagram</div>
            <div>{data?.socials.instagram}</div>
          </dl>
          </div>
        </aside>
        <hr/>
        <aside className='user-socials'>
        <h5>Guarantor</h5>
          <div>
          <dl>
            <div className='user-name'>full Name</div>
            <div>{data?.guarantor.firstName} {data?.guarantor.lastName}</div>
          </dl>
          <dl>
            <div className='user-name'>Phone Number</div>
            <div>{data?.guarantor.phoneNumber}</div>
          </dl>
          <dl>
            <div className='user-name'> Email Address</div>
            <div>debby@gmail.com</div>
          </dl>
          <dl>
            <div className='user-name'> Relationship</div>
            <div>Sister</div>
          </dl>
          </div>
        </aside>
        <hr/>
        <aside className='user-socials'>
        <h5>&nbsp;</h5>
          <div>
          <dl>
            <div className='user-name'>full Name</div>
            <div>Debby Ogana</div>
          </dl>
          <dl>
            <div className='user-name'>Phone Number</div>
            <div>07060780922</div>
          </dl>
          <dl>
            <div className='user-name'> Email Address</div>
            <div>debby@gmail.com</div>
          </dl>
          <dl>
            <div className='user-name'> Relationship</div>
            <div>Sister</div>
          </dl>
          </div>
        </aside>

        
      </section>
    </section>
    </div>
  )
}

export default Userdetails
