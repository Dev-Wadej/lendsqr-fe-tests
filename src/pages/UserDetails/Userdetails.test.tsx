

import Userdetails from './Userdetails'
import renderer from 'react-test-renderer';
import {cleanup,} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ApiProvider } from '@reduxjs/toolkit/dist/query/react';
import { apiSlice } from '../../features/api/apiSlice';



afterEach(cleanup)
test('Userdetails component snapshot HTML', () => {
    const output = renderer.create(<MemoryRouter>
        <ApiProvider api={apiSlice}>
       <Userdetails/>
        </ApiProvider>
    </MemoryRouter>  ).toJSON();
  expect(output).toMatchSnapshot();
});