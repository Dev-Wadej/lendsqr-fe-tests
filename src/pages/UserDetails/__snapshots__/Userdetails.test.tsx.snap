// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Userdetails component snapshot HTML 1`] = `
<div
  className="user-details-section-wrapper"
>
  <div
    className="user-hamburger"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <section
    className="user-details-section"
  >
    <div
      className="user-details-arrow-back"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <span>
        <svg>
          backarrow.svg
        </svg>
      </span>
      <span>
        Back to Users
      </span>
    </div>
    <div
      className="user-details-header"
    >
      <h3>
        User Details
      </h3>
      <dl>
        <button>
          BLACKLIST USER
        </button>
        <button>
          ACTIVATE USER
        </button>
      </dl>
    </div>
    <div
      className="user-top-details-wrapper"
    >
      <section
        className="user-details-top"
      >
        <aside
          className="user-details-avatar-wrapper"
        >
          <span
            className="avatar-img"
          >
            <img
              alt="User Avatar"
            />
          </span>
          <div>
            <h4>
               
            </h4>
            <span />
          </div>
        </aside>
        <aside
          className="user-tier"
        >
          <h3>
            User's Tier
          </h3>
          <span>
            <svg>
              starfilled.svg
            </svg>
            <svg>
              starunfilled.svg
            </svg>
            <svg>
              starunfilled.svg
            </svg>
          </span>
        </aside>
        <aside
          className="user-numerics"
        >
          <h4>
            ₦200,000.00
          </h4>
          <p>
            9912345678/Providus Bank
          </p>
        </aside>
      </section>
      <section
        className="user-tab"
      >
        <button
          className="active"
        >
          General Details
        </button>
        <button>
          Documents
        </button>
        <button>
          Bank Details
        </button>
        <button>
          Loans
        </button>
        <button>
          Savings
        </button>
        <button>
          App and System
        </button>
      </section>
    </div>
    <section
      className="user-heavy-info"
    >
      <aside
        className="user-personal-info"
      >
        <h5>
          Personal Information
        </h5>
        <div>
          <dl>
            <div
              className="user-name"
            >
              full Name
            </div>
            <div>
               
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Phone Number
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Email Address
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Bvn
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Gender
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Marital status
            </div>
            <div>
              Single
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Children
            </div>
            <div>
              None
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Type of residence
            </div>
            <div>
              Parent’s Apartment
            </div>
          </dl>
        </div>
      </aside>
      <hr />
      <aside
        className="user-education-job"
      >
        <h5>
          Education and Employment
        </h5>
        <div>
          <dl>
            <div
              className="user-name"
            >
              level of education
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              employment status
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              sector of employment
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Duration of employment
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              office email
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Monthly income 
            </div>
            <div>
              ₦
               - ₦
               
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              loan repayment
            </div>
            <div />
          </dl>
        </div>
      </aside>
      <hr />
      <aside
        className="user-socials"
      >
        <h5>
          Socials
        </h5>
        <div>
          <dl>
            <div
              className="user-name"
            >
              Twitter
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Facebook
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
               Instagram
            </div>
            <div />
          </dl>
        </div>
      </aside>
      <hr />
      <aside
        className="user-socials"
      >
        <h5>
          Guarantor
        </h5>
        <div>
          <dl>
            <div
              className="user-name"
            >
              full Name
            </div>
            <div>
               
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Phone Number
            </div>
            <div />
          </dl>
          <dl>
            <div
              className="user-name"
            >
               Email Address
            </div>
            <div>
              debby@gmail.com
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
               Relationship
            </div>
            <div>
              Sister
            </div>
          </dl>
        </div>
      </aside>
      <hr />
      <aside
        className="user-socials"
      >
        <h5>
           
        </h5>
        <div>
          <dl>
            <div
              className="user-name"
            >
              full Name
            </div>
            <div>
              Debby Ogana
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
              Phone Number
            </div>
            <div>
              07060780922
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
               Email Address
            </div>
            <div>
              debby@gmail.com
            </div>
          </dl>
          <dl>
            <div
              className="user-name"
            >
               Relationship
            </div>
            <div>
              Sister
            </div>
          </dl>
        </div>
      </aside>
    </section>
  </section>
</div>
`;
